Zend/zend_ast.c

```diff
@@ -441,3 +444,1011 @@ ZEND_API void zend_ast_apply(zend_ast *ast, zend_ast_apply_func fn) {
+				zend_ast_export_indent(str, indent);
+				smart_str_appends(str, "}\n");
+			} else {
+				smart_str_appendc(str, ';');
+			}
+			break;
+		case ZEND_AST_USE_ELEM:
+		case ZEND_AST_TRAIT_ALIAS:
+			zend_ast_export_name(str, ast->child[0], 0, indent);
+			if (ast->child[1]) {
+				smart_str_appends(str, " as ");
+				zend_ast_export_name(str, ast->child[1], 0, indent);
+			}
+			break;
+
+		/* 3 child nodes */
+		case ZEND_AST_METHOD_CALL:
+			zend_ast_export_ex(str, ast->child[0], 0, indent);
+			smart_str_appends(str, "->");
+			zend_ast_export_var(str, ast->child[1], 0, indent);
+			smart_str_appendc(str, '(');
+			zend_ast_export_ex(str, ast->child[2], 0, indent);
+			smart_str_appendc(str, ')');
+			break;
+		case ZEND_AST_STATIC_CALL:
+			zend_ast_export_name(str, ast->child[0], 0, indent);
+			smart_str_appends(str, "::");
+			zend_ast_export_var(str, ast->child[1], 0, indent);
+			smart_str_appendc(str, '(');
+			zend_ast_export_ex(str, ast->child[2], 0, indent);
+			smart_str_appendc(str, ')');
+			break;
+		case ZEND_AST_CONDITIONAL:
+			if (priority > 100) smart_str_appendc(str, '(');
+			zend_ast_export_ex(str, ast->child[0], 100, indent);
+			if (ast->child[1]) {
+				smart_str_appends(str, " ? ");
+				zend_ast_export_ex(str, ast->child[1], 101, indent);
+				smart_str_appends(str, " : ");
+			} else {
+				smart_str_appends(str, " ?: ");
+			}
+			zend_ast_export_ex(str, ast->child[2], 101, indent);
+			if (priority > 100) smart_str_appendc(str, ')');
+			break;
+
+		case ZEND_AST_TRY:
+			smart_str_appends(str, "try {\n");
+			zend_ast_export_stmt(str, ast->child[0], indent + 1);
+			zend_ast_export_indent(str, indent);
+			zend_ast_export_ex(str, ast->child[1], 0, indent);
+			if (ast->child[2]) {
+				smart_str_appends(str, "} finally {\n");
+				zend_ast_export_stmt(str, ast->child[2], indent + 1);
+				zend_ast_export_indent(str, indent);
+			}
+			smart_str_appendc(str, '}');
+			break;
+		case ZEND_AST_CATCH:
+			smart_str_appends(str, "} catch (");
+			zend_ast_export_ex(str, ast->child[0], 0, indent);
+			zend_ast_export_ex(str, ast->child[1], 0, indent);
+			smart_str_appends(str, ") {\n");
+			zend_ast_export_stmt(str, ast->child[2], indent + 1);
+			zend_ast_export_indent(str, indent);
+			break;
+		case ZEND_AST_PARAM:
+			if (ast->child[0]) {
+				zend_ast_export_name(str, ast->child[0], 0, indent);
+				smart_str_appendc(str, ' ');
+			}
+			if (ast->attr) {
+				smart_str_appendc(str, '&');
+			}
+			smart_str_appendc(str, '$');
+			zend_ast_export_name(str, ast->child[1], 0, indent);
+			APPEND_DEFAULT_VALUE(2);
+
+		/* 4 child nodes */
+		case ZEND_AST_FOR:
+			smart_str_appends(str, "for (");
+			zend_ast_export_ex(str, ast->child[0], 0, indent);
+			smart_str_appends(str, "; ");
+			zend_ast_export_ex(str, ast->child[1], 0, indent);
+			smart_str_appends(str, "; ");
+			zend_ast_export_ex(str, ast->child[2], 0, indent);
+			smart_str_appends(str, ") {\n");
```

Comment:
Not really important, but this will have weird formatting in the `for (;;)` case.